Documentation on how to add asset through the UI (#127)
agera-edc/MinimumViableDataspace: docs/developer/add-asset.md

# Add Asset

## Overview

> A new asset can be added through MVD UI, in this process we actually point to an pre-existing asset in participant's respective object storage account. In this version of MVD, only Azure Object Storage is supported.

Reviewer: ```suggestion
A new asset can be added through the EDC data dashboard, in this process we actually point to an pre-existing asset in participant's respective object storage account. In this version of MVD, only Azure Object Storage is supported.
```

> A new asset can be added through the EDC data dashboard, in this process we actually point to an pre-existing asset in participant's respective object storage account. In this version of MVD, only Azure Object Storage is supported.

Reviewer 2: nit: This kind of information (`In this version`) is a good candidate to get outdated. IMHO, a code comment in the respective UI code which fills the supported storages would be helpful. The code comment should state that as soon more than the `Azure Object Storage` is supported, this doc file needs to get updated.

(I have not checked that code. If it is already there, ignore this comment.)

Author: So leaving it open for now as to me seems like we will be having more feature supported with UI later and not in this phase of MVD.

> So for instance if we want to add and asset named as `test-document` then this asset must be already available in the participant's object storage account under container named as `src-container` before we can add it via MVD UI. Refer to the following links for more details about how to upload an asset in Azure Object Storage:

Reviewer: ```suggestion
So for instance if we want to add and asset named as `test-document` then this asset must be already available in the participant's object storage account under container named as `src-container` before we can add it via the dashboard UI. Refer to the following links for more details about how to upload an asset in Azure Object Storage:
```

> So for instance if we want to add and asset named as `test-document` then this asset must be already available in the participant's object storage account under container named as `src-container` before we can add it via the dashboard UI. Refer to the following links for more details about how to upload an asset in Azure Object Storage:

Reviewer 2: nit: typo & formulation

```suggestion
So for instance if we want to add an asset named as `test-document` then this asset must be already available in the participant's object storage account in a container named as `src-container` before we can add it via the dashboard UI. Refer to the following links for more details about how to upload an asset in Azure Object Storage:
```

* [Create a container](https://docs.microsoft.com/azure/storage/blobs/storage-quickstart-blobs-portal#create-a-container)
* [Upload a blob](https://docs.microsoft.com/azure/storage/blobs/storage-quickstart-blobs-portal#upload-a-block-blob)

## Prerequisites

> Actual asset must be already uploaded in object storage account under container named as `src-container`.

Reviewer 2: nit: formulation

```suggestion
Actual asset must be already uploaded in object storage account in a container named as `src-container`.
```

> ## Adding an asset

Author: Yes, good. Applied.

* Go to`Assets` pane and here you will see all existing assets if any.
* Click on `Create asset` button and this will open an UI popup where you can add further details.

> * Add a unique ID e.g. UUID for the asset in text box named `id`.

Reviewer 3: It would be good to add a description for what the ID will be used.

* Add name for the asset in text box named `name`.

> * Add content type for the asset in text box named `Content Type`. e.g. `text/plain`

Reviewer 3: We could add a description why the content type has to be defined and for what it is used.

Author: Updated as discussed offline. That this is just a metadata for now and not used further in data transfers.

> * `Azure Storage` must be pre-selected in the dropdown named as `Destination`.

Reviewer 3: ```suggestion
* Select the `Destination` type (`Azure Storage` in this case to transfer from Azure Blob Storage to Azure Blob Storage).
```

Author: Applied.

> * `Azure Storage` must be pre-selected in the dropdown named as `Destination`.

Reviewer 2: sounds like precondition, no instruction

```suggestion
* `Azure Storage` is already pre-selected in the dropdown named as `Destination`.
```

* Add object storage account name in text box named as `Account`.

> * `src-container` must pre-filled in the text box named as `Container`.

Reviewer 3: ```suggestion
* Define the container within the Azure Object Storage Account in `Container` (pre-defined as `src-container` for the MVD for simplicity reasons).
```
